velox/vector/tests/DecodedVectorTest.cpp

```diff
@@ -122,9 +143,45 @@ class DecodedVectorTest : public testing::Test, public VectorTestBase {
     assertDecodedVector(data, flatVector.get(), false);
   }
 
+  template <typename T>
+  void testFlatDecimals(const TypePtr& type) {
+    auto cardinality = 10010;
+    auto cardData = genTestData<T>(
+        cardinality,
+        /* includeNulls */ true,
+        false /* sorted */,
+        false /* useFullTypeRange */,
+        std::nullopt /* seed */,
+        std::nullopt /* fixedWidthStringSize */,
+        type);
+    const auto& data = cardData.data();
+    EXPECT_EQ(cardinality, data.size());
+
+    VectorPtr decimalFlat;
+    if constexpr (std::is_same_v<T, UnscaledShortDecimal>) {
```

Comment:
this can go into `testFlat`? Why do we need `testFlatDecimal`?

Reply:
Done.